test/test_simexec.cpp

```diff
@@ -0,0 +1,65 @@
+#include "CircuitSimulatorExecutor.hpp"
+#include "SimulationTask.hpp"
+#include "TestHelpers.hpp"
+
+#include "gtest/gtest.h"
+
+struct TestConfigurationDDSIM {
+  // given input
+  std::string description;
+  std::string initialCircuit;
+
+  // expected output
+  std::size_t expected11{};
+};
+
+// NOLINTNEXTLINE (readability-identifier-naming)
+inline void from_json(const nlohmann::json& j, TestConfigurationDDSIM& test) {
+  test.description    = j.at("description").get<std::string>();
+  test.initialCircuit = j.at("initial_circuit").get<std::string>();
+  test.expected11     = j.at("expected_11").get<std::size_t>();
+}
+
+static std::vector<TestConfigurationDDSIM> getTests(const std::string& path) {
+  std::ifstream  input(path);
+  nlohmann::json j;
+  input >> j;
+  return j;
+}
+
+class DDSIMExecTest : public ::testing::TestWithParam<TestConfigurationDDSIM> {
+protected:
+  void SetUp() override {
+    test = GetParam();
+
+    std::stringstream      ss(test.initialCircuit);
+    qc::QuantumComputation qc{};
+    qc.import(ss, qc::Format::OpenQASM);
+    std::cout << "Circuit:\n" << qc;
+
+    qcSim          = std::make_unique<qc::QuantumComputation>(qc.clone());
+    simulationTask = std::make_unique<SimulationTask>(std::move(qcSim));
```

Comment:
This kind of construction does not look efficient and/or convenient at all.
```suggestion
    std::stringstream      ss(test.initialCircuit);
    auto qc = std::make_unique<qc::QuantumComputation>():
    qc->import(ss, qc::Format::OpenQASM);
    std::cout << "Circuit:\n" << qc;

    simulationTask = std::make_unique<SimulationTask>(std::move(qc));
```
completely eliminates that one copy.
Furthermore, the simulation task doesn't need to be a unique_ptr. It can just be a regular, plain old `SimulationTask`